XenAdmin/Wizards/PatchingWizard/PatchingWizard_PrecheckPage.cs

```diff
@@ -470,6 +470,22 @@ protected virtual List<CheckGroup> GenerateCommonChecks(List<Host> applicableSer
                     new List<Check> { new XenCenterVersionCheck(highestNewVersion ?? UpdateAlert.NewServerVersion) }));
             }
 
+            //PVGuestsCheck checks
+            var pvChecks = new List<Check>();
+            foreach (Host host in applicableServers.Where(h => !Helpers.QuebecOrGreater(h.Connection)))
+             {
+                if (host.GetPvVMs() != null && host.GetPvVMs().Count > 0)
+                {
+                    pvChecks.Add(new PVGuestsCheck(host)); 
+                    break;
+                }   
+            }
+             if (pvChecks.Count > 0)
+             {
+                 groups.Add(new CheckGroup(Messages.CHECKING_PV_GUESTS, pvChecks));
+             }
+            
+

```

Comment:
This check should only run if we're updating to a new version. Currently it's run for any hotfix or suppack the user applies.

Reply:
Also, the content of the check (and the same in the RPU precheck page) can be written more efficiently. First of all, we need to decide whether it's a pool or host check. If we assume that it is a pool check (because this is what the logic in GetPvVMs() says) you can write it as:
```
var pvChecks = new List<Check>();
foreach (Host host in SelectedMasters.Where(h => !Helpers.QuebecOrGreater(h.Connection)))
{
    if (host.Connection.Cache.VMs.Any(vm => vm.IsPvVm()))
        pvChecks.Add(new PVGuestsCheck(host));
}

if (pvChecks.Count > 0)
    groups.Add(new CheckGroup(Messages.CHECKING_PV_GUESTS, pvChecks));
```